server/controllers/medical/patient.js

```diff
@@ -429,3 +452,148 @@ function logVisit(patientData, userId) {
+      'q.account_id, q.price_list_uuid, q.is_convention, q.locked ' +
+    'FROM (' +
+      'SELECT p.uuid, p.project_id, CONCAT(proj.abbr, p.reference) AS reference, p.debitor_uuid, p.creditor_uuid, ' +
+        'p.first_name, p.last_name, p.middle_name, p.sex, p.dob, p.father_name, p.mother_name, ' +
+        'p.profession, p.employer, p.spouse, p.spouse_profession, p.spouse_employer, ' +
+        'p.religion, p.marital_status, p.phone, p.email, p.address_1, p.address_2, ' +
+        'p.renewal, p.origin_location_id, p.current_location_id, p.registration_date, ' +
+        'p.title, p.notes, p.hospital_no, d.text, ' +
+        'dg.account_id, dg.price_list_uuid, dg.is_convention, dg.locked ' +
+      'FROM patient AS p JOIN project AS proj JOIN debitor AS d JOIN debitor_group AS dg ' +
+        'ON p.debitor_uuid = d.uuid AND d.group_uuid = dg.uuid AND p.project_id = proj.id' +
+    ') AS q ';
+
+    var clauseWhere = 'WHERE ';
+
+    data = [];
+
+    var criteria = Object.keys(qFields).map(function (item) {
+      data.push(qFields[item])
+      return 'q.' + item + ' = ?';
+    }).join(' AND ');
+
+    clauseWhere += criteria;
+    sql += clauseWhere;
+
+  } else if (qReference) {
+    // find a patient identified by a reference. Ex. HBB123
+    sql =
+      'SELECT q.uuid, q.project_id, q.reference, q.debitor_uuid, q.creditor_uuid, ' +
+        'q.first_name, q.last_name, q.middle_name, q.sex, q.dob, q.father_name, q.mother_name, ' +
+        'q.profession, q.employer, q.spouse, q.spouse_profession, q.spouse_employer, ' +
+        'q.religion, q.marital_status, q.phone, q.email, q.address_1, q.address_2, ' +
+        'q.renewal, q.origin_location_id, q.current_location_id, q.registration_date, ' +
+        'q.title, q.notes, q.hospital_no, q.text, ' +
+        'q.account_id, q.price_list_uuid, q.is_convention, q.locked ' +
+      'FROM (' +
+        'SELECT p.uuid, p.project_id, CONCAT(proj.abbr, p.reference) AS reference, p.debitor_uuid, p.creditor_uuid, ' +
+          'p.first_name, p.last_name, p.middle_name, p.sex, p.dob, p.father_name, p.mother_name, ' +
+          'p.profession, p.employer, p.spouse, p.spouse_profession, p.spouse_employer, ' +
+          'p.religion, p.marital_status, p.phone, p.email, p.address_1, p.address_2, ' +
+          'p.renewal, p.origin_location_id, p.current_location_id, p.registration_date, ' +
+          'p.title, p.notes, p.hospital_no, d.text, ' +
+          'dg.account_id, dg.price_list_uuid, dg.is_convention, dg.locked ' +
+        'FROM patient AS p JOIN project AS proj JOIN debitor AS d JOIN debitor_group AS dg ' +
+          'ON p.debitor_uuid = d.uuid AND d.group_uuid = dg.uuid AND p.project_id = proj.id' +
+      ') AS q ' +
+      'WHERE q.reference = ? ';
+
+      data = [qReference];
+
+  } else {
+    // throw an error in other cases
+    throw req.codes.ERR_PARAMETERS_REQUIRED;
+  }
+
+  if (qLimit && typeof(qLimit) === 'number' && (parseFloat(qLimit) === parseInt(qLimit))) {
```

Comment:
Instead of doing this equality check, you might just try this:
```js
if (qLimit && typeof qLimit === 'number') {
  sql += 'LIMIT ' + Math.floor(qLimit) + ';';
}
```
The result is that "limit=12.134" will result in the same thing as "limit=12", but this seems acceptable.